cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -202,12 +202,16 @@ func (s *CMServer) Run(_ []string) error {
 	}
 
 	if s.AllocateNodeCIDRs {
-		routes, ok := cloud.Routes()
-		if !ok {
-			glog.Fatal("Cloud provider must support routes if allocate-node-cidrs is set")
+		if cloud != nil {
```

Comment:
I think it'd also make sense to just warn and then run in allocate-only mode for clouds that don't support the Routes interface.
To clean up the if block a little bit, how about:
```
if s.AllocateNodeCIDRs {
	if cloud == nil {
		glog.Warningf("allocate-node-cidrs is set, but no cloud provider specified. Will not manage routes.")
	} else if routes, ok := cloud.Routes(); !ok {
		glog.Warningf("allocate-node-cidrs is set, but cloud provider does not support routes. Will not manage routes.")
	} else {
		routeController := routecontroller.New(routes, kubeClient, s.ClusterName, (*net.IPNet)(&s.ClusterCIDR))
		routeController.Run(s.NodeSyncPeriod)
	}
}
```

It might violate Go's "only indent the exceptional case" rule, but I think it's reasonable here where there isn't necessarily an "exceptional" case, and it looks way cleaner to me.